python/cudf/cudf/testing/_utils.py

```diff
@@ -376,16 +376,17 @@ class cudf_timeout:
     Context manager to raise a TimeoutError after a specified number of seconds.
     """
 
-    def __init__(self, seconds, *, timeout_message=""):
-        self.seconds = int(seconds)
-        self.timeout_message = timeout_message
-
-    def _timeout_handler(self, signum, frame):
-        raise TimeoutError(self.timeout_message)
+    def __init__(self, timeout):
+        self.timeout = int(timeout)
 
     def __enter__(self):
-        signal.signal(signal.SIGALRM, self._timeout_handler)
-        signal.alarm(self.seconds)
-
-    def __exit__(self, type, value, traceback):
-        signal.alarm(0)
+        self.start_time = time.perf_counter()
+
+    def __exit__(self):
+        elapsed_time = int(
```

Comment:
Why cast to int? This will round the value down unnecessarily.

Reply:
Removed the cast to int. I wanted to keep the functionality limited to seconds (int) as `.perf_counter` returns float.